ccloader/js/filemanager.js

```diff
@@ -51,40 +56,25 @@ export class Filemanager {
 	}
 
 	getAllModsFiles() {
-		const subs = this._getModFoldersOnAndroid('/') || this._getFolders();
+		const subs = isAndroid && this._androidModFolders ? this._androidModFolders : this._getFolders();
 		return [].concat(...subs.map(sub => this._getResourcesInFolder(sub, this.pathSep + 'package.json')));
 	}
 
 	getAllCCModFiles() {
-		const subs = this._getModFoldersOnAndroid('/') || this._getFolders();
+		const subs = isAndroid && this._androidModFolders ? this._androidModFolders : this._getFolders();
 		return [].concat(...subs.map(sub => this._getResourcesInFolder(sub, this.pathSep + 'ccmod.json')));
 	}
 
 	getAllModPackages() {
-		return this._getModFoldersOnAndroid('.ccmod') || this._getResourcesInFolder(null, '.ccmod');
-	}
-
-	_getModFoldersOnAndroid(ending) {
-		const provider = window.CrossAndroidModListProvider;
-		if (provider && provider.getModListAsJson) {
-			const entries = [];
-			for (let entry of JSON.parse(provider.getModListAsJson())) {
-				if (entry.endsWith(ending)) {
-					if (ending === '/') {
-						entry = entry.slice(0, -1);
-					}
-					entries.push('assets/mods/' + entry);
-				}
-			}
-			return entries;
+		if (isAndroid && this._androidModPackages) {
+			return this._androidModPackages;
 		}
-		return null;
+		return this._getResourcesInFolder(null, '.ccmod');
 	}
 
 	getExtensions() {
-		const provider = window.CrossAndroidExtensionListProvider;
-		if (provider && provider.getExtensionListAsJson) {
-			return JSON.parse(provider.getExtensionListAsJson());
+		if (isAndroid && this._androidExtensions) {

```

Comment:
The extra check for `_androidExtensions` is a bit strange but works

Reply:
Take a look at the `_receiveInfoFromAndroid` function, it only calls the `get{Extension,Mod}ListAsJson` functions if they are really available. I wrote it that way for the case if we wanted to change the APIs in the future.